evm/src/generation/mpt.rs

```diff
@@ -89,13 +120,13 @@ pub struct AccessListTransactionRlp {
     pub to: AddressOption,
     pub value: U256,
     pub data: Bytes,
-    pub access_list: Vec<AccessListItemRlp>,
+    pub access_list: Vec<AccessListInner>,
```

Comment:
I'm a bit confused by this. EIP-2930 says that the format for `access_list` is `[[{20 bytes}, [{32 bytes}...]]...]`, i.e., `Vec<AccessListItemRlp>`. But this change seems to indicate that it could also be something like `Vec<Vec<AccessListItemRlp>>`.
Are there txns with this access list format?

Reply:
I agree, this doesn't look right: like it stands this would be accepted: `vec![AccessListInner::List(vec![AccessListItemRlp{address: …, storage_keys: …}, AccessListItemRlp{address: …, storage_keys: …}])]`, which seems odd.

Reply:
I agree that this isn't the regular encoding, but the EVM test suite has some weird block RLP encodings on some of its tests, which either need to be ignored when parsing or require this specific handling.
The other weird encoding is at the transactions section of the block, I've seen so far three versions:

- a list of txns
- a single txn
- a list of strings (ie doubly encoded)

Reply:
What I could do then, if just port these changes directly to evm-tests instead. I need some logic there anyway to handle the non-RLP prefix for those transactions, so it wouldn't add too much, and at least here we'd respect the specs.

Reply:
Yeah I think that's probably a better idea. These are public structs so I feel like we should have them follow the specs and handle the test suite's oddities in the `evm-tests` repo.

Reply:
Good idea 👍